### PR TITLE
[WIP] Add virtctl autocomplete support for bash and zsh

### PR DESCRIPTION
**Pull request by rmohr**
Type "virtctl completion --help" to find out on how to install bash and
zsh completion for virtctl.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

**Comment by fabiand** [comment]
Will you also add some functional tests?

File: pkg/virtctl/completion/completion.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/spf13/cobra"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+func NewCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "completion SHELL",
+		Short: "Output shell completion code for the specified shell (bash or zsh)",
+		Example: `# Load the virtctl completion code for bash into the current shell
```
Comment (stu-gott) [inline pkg/virtctl/completion/completion.go]: Would it be worth mentioning that by convention these sorts of completion functions tend to be placed in `/etc/bash_completion.d/`? Not sure if that starts to get outside the scope of these instructions though...

File: pkg/virtctl/console/console.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ import (
 
 func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "console (vm)",
+		Use:     "console VM",
```
Comment (davidvossel) [inline pkg/virtctl/console/console.go]: To me, it seemed more obvious that vm is an argument the previous way this was documented.

Comment (rmohr) [reply pkg/virtctl/console/console.go]: Hm, to use the same structure like kubectl, it would have to be `console VM_NAME`.

**Comment by rmohr** [comment]
I still think this has a lot of value, especially in combination with #1361

/remove-lifecycle stale

**Comment by SchSeba** [comment]
I still think this has a lot of value

/remove-lifecycle stale

**Comment by SchSeba** [comment]
@fabiand I hope to have some time to take a look into that next week.

**Comment by rmohr** [comment]
/close

If we add this, it is superseeded by #2015

**Pull request closed without merging**